src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -829,14 +842,13 @@ checkFunctionApplication' fn (ConstrainedType _ con fnTy) arg = do
 checkFunctionApplication' fn fnTy dict@TypeClassDictionary{} =
   return (fnTy, App fn dict)
 checkFunctionApplication' fn u arg = do
-  arg' <- do
-    TypedValue _ arg' t <- infer arg
+  TypedValue' checkType expr ty <- do
```

Comment:
Oops - I just noticed this, which is little a bit awkward now, I think. How about using a pattern `tv@TypedValue' _ _ ty` here, and then using `tvToExpr tv` below instead of the `TypedValue` constructor?